### PR TITLE
4498 - Add automation id for Splitter, Switch, Timeline, Timepicker and Signin

### PR DESCRIPTION
**Pull request by ericangeles**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Adds automation id for splitter, switch, timeline, timepicker and signin components. Also includes docs and e2e.

**Related github/jira issue (required)**:

Partial https://github.com/infor-design/enterprise/issues/4498

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app

1.) Splitter
- Go to http://localhost:4000/components/splitter/example-index
- Open developers tool and inspect the element
- `splitter`, `splitter handle` and `splitter icon` should all have id and automation id.
- NOTE: the id of handle and icon should append it's name. e.g `data-automation-id='splitter-automation-id-1-handle'`, `data-automation-id='splitter-automation-id-1-icon'`

2.) Timepicker
- Go to http://localhost:4000/components/timepicker/example-index
- Open developers tool and inspect the element
- `timepicker` and `timepicker trigger` should have id and automation id.
- The `trigger` should append its name on the id. e.g. `timepicker-automation-id-1-trigger`
- Click on timepicker trigger
- Inspect the element on dropdowns `hours`, `minutes`, `seconds`, `period` and `set time`
- It should all have ids and automation ids

3.) Signin, Switch, & Timeline
- Go to http://localhost:4000/components/signin/example-index
- Go to http://localhost:4000/components/switch/example-index
- Go to http://localhost:4000/components/timeline/example-index
- Inspect the element
- The main component should have id and automation id
- These three does not need special cases like splitter and timepicker since you can add ids manually

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: app/views/components/signin/example-index.html [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
+    <section class="signin" role="main" id="signin-container-id">
 
         <svg class="icon icon-logo" focusable="false" aria-hidden="true" role="presentation">
           <use href="#icon-logo"></use>
         </svg>
 
         <h1>Application Name</h1>
 
-        <form id="signin" data-validate-on="submit" autocomplete="off">
+        <form id="signin" data-validate-on="submit" autocomplete="off" data-automation-id="signin-automation-id">
```
Comment (tmcconechy) [inline app/views/components/signin/example-index.html:14]: Might want to show data-automation-id on all the inputs

**Review by tmcconechy — APPROVED** [review]

**Review by EdwardCoyle — CHANGES_REQUESTED** [review]
Just a couple things:
- comments below
- I was getting a JS console error on http://localhost:4000/components/splitter/example-index, however the automation ids are there.  It looks like the splitter is being updated with new settings, and the `destroy().init()` chain in there isn't completing... might be a separate issue?

File: src/components/timepicker/timepicker.js [hunk 2]
```diff
@@ -213,6 +215,16 @@ TimePicker.prototype = {
...
   },
 
+  /**
+   * Adds an extra attributes
+   * @private
+   * @returns {void}
+   */
+  addAttributes() {
+    utils.addAttributes(this.element, this, this.settings.attributes);
+    utils.addAttributes(this.trigger, this, this.settings.attributes, 'trigger');
```
Comment (EdwardCoyle) [inline src/components/timepicker/timepicker.js:225]: I think we probably also want to have the individual Dropdowns get ids as well (hours, mins, seconds, day period)

**Comment by ericangeles** [comment]
@EdwardCoyle yeah, and it seems that the CI does not detect this error though I have error detection on e2e. Yep, a separate issue will do.

**Review by EdwardCoyle — APPROVED** [review]

**Review by tmcconechy — APPROVED** [review]
Probably just need a test like https://github.com/infor-design/enterprise/blob/master/test/components/about/about.e2e-spec.js#L22 inside splitter but most important is to fix

**Comment by deep7102** [comment]
looks good, just one thing see the js error http://localhost:4000/components/splitter/example-index

image

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Just fix that splitter error then maybe add a test as i mentioned 👍

**Review by jmacaluso711 — DISMISSED** [review]

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -43,6 +43,7 @@
 - `[Rating/Slider/Spinbox]` Added `attributes` setting to set automation id's and id's. ([#4498](https://github.com/infor-design/enterprise/issues/4498))
 - `[Textarea]` Added `attributes` setting to set automation ids's and id's. ([#4498](https://github.com/infor-design/enterprise/issues/4498))
 - `[Treemap]` Added ability to show a tooltip. ([#2794](https://github.com/infor-design/enterprise/issues/2794))
+- `[Splitter/Switch/Timeline/Timepicker/Signin]` Added `attributes` setting to set automation ids's and id's. ([#4498](https://github.com/infor-design/enterprise/issues/4498))
```
Comment (tmcconechy) [inline docs/CHANGELOG.md:46]: Can remove this as i made one entry for all components https://github.com/infor-design/enterprise/pull/4554/files

**Review by tmcconechy — APPROVED** [review]

**Review by jmacaluso711 — APPROVED** [review]

**Pull request merged**